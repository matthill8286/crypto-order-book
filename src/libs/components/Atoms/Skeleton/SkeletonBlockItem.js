import React from 'react';
import { pulse } from "../../../styles/animation";
import { css, styled } from "../../../styles/styled";
export var StyledSkeletonBlockItem = styled.div.withConfig({
  displayName: "SkeletonBlockItem__StyledSkeletonBlockItem",
  componentId: "sc-119erzb-0"
})(function (_ref) {
  var width = _ref.width,
      height = _ref.height,
      marginTop = _ref.marginTop,
      _ref$animated = _ref.animated,
      animated = _ref$animated === void 0 ? false : _ref$animated;
  return css(["position:relative;background-color:rgba(0,0,0,.1);", " ", " ", " ", ""], animated && css(["", ";"], pulse), marginTop && "margin-top: ".concat(marginTop, ";"), width && "width: ".concat(width, ";"), height && "height: ".concat(height, ";"));
});
export var SkeletonBlockItem = function SkeletonBlockItem() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {
    width: 'auto'
  };
  return React.createElement(StyledSkeletonBlockItem, props);
};
//# sourceMappingURL=SkeletonBlockItem.js.map