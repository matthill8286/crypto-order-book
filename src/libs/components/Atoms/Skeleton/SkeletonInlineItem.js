function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { pulse } from "../../../styles/animation";
import { css, styled } from "../../../styles/styled";
import { Typo } from "../Typography";
export var StyledSkeletonInlineItem = styled(Typo).withConfig({
  displayName: "SkeletonInlineItem__StyledSkeletonInlineItem",
  componentId: "cxlfm1-0"
})(["", " ", " color:rgba(0,0,0,0);background-color:rgba(0,0,0,0.1);width:unset;&::before{content:'';display:block;}"], function (_ref) {
  var _ref$animated = _ref.animated,
      animated = _ref$animated === void 0 ? false : _ref$animated;
  return animated && css(["", ";"], pulse);
}, function (_ref2) {
  var tag = _ref2.tag,
      newLine = _ref2.newLine;
  return !tag && "display: ".concat(newLine ? 'table' : 'inline-block', ";");
});
export var SkeletonInlineItem = function SkeletonInlineItem(_ref3) {
  var _ref3$length = _ref3.length,
      length = _ref3$length === void 0 ? 12 : _ref3$length,
      text = _ref3.text,
      props = _objectWithoutProperties(_ref3, ["length", "text"]);

  return React.createElement(StyledSkeletonInlineItem, props, text || 'a'.repeat(length));
};
//# sourceMappingURL=SkeletonInlineItem.js.map