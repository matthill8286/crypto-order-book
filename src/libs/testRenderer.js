import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map