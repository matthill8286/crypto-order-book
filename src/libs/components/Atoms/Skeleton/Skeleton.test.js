import React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { SkeletonBlockItem } from "./SkeletonBlockItem";
import { SkeletonInlineItem } from "./SkeletonInlineItem";
describe('Skeleton', function () {
  describe('SkeletonBlockItem', function () {
    it('should render', function () {
      var wrapper = renderWithTheme(React.createElement(SkeletonBlockItem, {
        height: "200px"
      }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render with height', function () {
      var wrapper = renderWithTheme(React.createElement(SkeletonBlockItem, {
        height: "200px",
        width: "100%"
      }));
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('SkeletonInlineItem', function () {
    it('should render', function () {
      var wrapper = renderWithTheme(React.createElement(SkeletonInlineItem, {
        text: "Lorem Ipsum"
      }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render with font-size', function () {
      var wrapper = renderWithTheme(React.createElement(SkeletonInlineItem, {
        text: "Lorem Ipsum",
        fontSize: "xxl"
      }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
//# sourceMappingURL=Skeleton.test.js.map