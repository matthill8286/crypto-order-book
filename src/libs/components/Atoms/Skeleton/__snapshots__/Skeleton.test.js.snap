// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton SkeletonBlockItem should render 1`] = `
<div
  className="SkeletonBlockItem__StyledSkeletonBlockItem-sc-119erzb-0 iCCOLN"
  height="200px"
/>
`;

exports[`Skeleton SkeletonBlockItem should render with height 1`] = `
<div
  className="SkeletonBlockItem__StyledSkeletonBlockItem-sc-119erzb-0 gMKmIW"
  height="200px"
  width="100%"
/>
`;

exports[`Skeleton SkeletonInlineItem should render 1`] = `
<span
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hSKBJn SkeletonInlineItem__StyledSkeletonInlineItem-cxlfm1-0 gKkfjf"
>
  Lorem Ipsum
</span>
`;

exports[`Skeleton SkeletonInlineItem should render with font-size 1`] = `
<span
  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 epOizM SkeletonInlineItem__StyledSkeletonInlineItem-cxlfm1-0 gKkfjf"
  fontSize="xxl"
>
  Lorem Ipsum
</span>
`;
